recommenders/models/newsrec/models/base_model.py

```diff
@@ -212,6 +214,8 @@ def fit(
             )
 
             for batch_data_input in tqdm_util:
+                if step_limit is not None and step_limit>=step_limit:
```

Comment:
```suggestion
                if step_limit is not None and step>=step_limit:
```
I think you mean here this? You were comparing step_limit with itself which is always the same